src/main/java/com/minecolonies/coremod/inventory/CraftingGUIBuilding.java

```diff
@@ -0,0 +1,314 @@
+     */
+    private static final int PLAYER_INVENTORY_COLUMNS = 9;
+
+    /**
+     * Initial x-offset of the inventory slot.
+     */
+    private static final int PLAYER_INVENTORY_INITIAL_X_OFFSET = 8;
+
+    /**
+     * Initial y-offset of the inventory slot.
+     */
+    private static final int PLAYER_INVENTORY_INITIAL_Y_OFFSET = 84;
+
+    /**
+     * Each offset of the inventory slots.
+     */
+    private static final int INVENTORY_OFFSET_EACH = 18;
+
+    /**
+     * Initial y-offset of the inventory slots in the hotbar.
+     */
+    private static final int PLAYER_INVENTORY_HOTBAR_OFFSET = 142;
+
+    /**
+     * The x position of the crafting result slot position.
+     */
+    private static final int X_CRAFT_RESULT = 124;
+
+    /**
+     * The y position of the crafting result slot position.
+     */
+    private static final int Y_CRAFT_RESULT = 35;
+
+    /**
+     * The x offset of the crafting slot position.
+     */
+    private static final int X_OFFSET_CRAFTING = 30;
+
+    /**
+     * The y offset of the crafting slot position.
+     */
+    private static final int Y_OFFSET_CRAFTING = 17;
+
+    /**
+     * Amount of slots in the crafting window.
+     */
+    private static final int CRAFTING_SLOTS = 5;
+
+    /**
+     * Amount of slots per line in the GUI.
+     */
+    private static final int SLOTS_PER_LINE = 9;
+
+    /**
+     * Start slot of the player hotbar.
+     */
+    private static final int HOTBAR_START = 32;
+
+    /**
+     * Total amount of slots in the GUI.
+     */
+    private static final int TOTAL_SLOTS = 41;
+
+    /**
+     * The crafting matrix inventory (2x2).
+     */
+    private final InventoryCrafting craftMatrix = new InventoryCrafting(this, 2, 2);
+
+    /**
+     * The crafting result slot.
+     */
+    private final IInventory craftResult = new InventoryCraftResult();
+
+    /**
+     * The world object.
+     */
+    private final World worldObj;
+
+    public CraftingGUIBuilding(final InventoryPlayer playerInventory, final World worldIn)
+    {
+        super();
+        this.worldObj = worldIn;
+
+        this.addSlotToContainer(new SlotCrafting(playerInventory.player, this.craftMatrix, this.craftResult, 0, X_CRAFT_RESULT, Y_CRAFT_RESULT) {
+            @Override
+            public boolean canTakeStack(final EntityPlayer playerIn)
+            {
+                return false;
+            }
+        });
+
+        for (int i = 0; i < 2; ++i)
+        {
+            for (int j = 0; j < 2; ++j)
+            {
+                this.addSlotToContainer(new Slot(this.craftMatrix, j + i * 2, X_OFFSET_CRAFTING + j * INVENTORY_OFFSET_EACH, Y_OFFSET_CRAFTING + i * INVENTORY_OFFSET_EACH){
+                    @Override
+                    public int getSlotStackLimit()
+                    {
+                        return 1;
+                    }
+
+                    @Override
+                    public ItemStack onTake(EntityPlayer p_190901_1_, ItemStack p_190901_2_)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)